src/ast_utils.h

```diff
@@ -46,6 +46,34 @@ struct BreakSeeker : public PostWalker<BreakSeeker, Visitor<BreakSeeker>> {
   }
 };
 
+// Finds all functions that are reachable via direct calls.
+
+struct DirectCallGraphAnalyzer : public PostWalker<DirectCallGraphAnalyzer, Visitor<DirectCallGraphAnalyzer>> {
+  Module *module;
+  std::vector<Function*> queue;
+  std::unordered_set<Function*> reachable;
+
+  DirectCallGraphAnalyzer(Module* module, std::vector<Function*> root) : module(module) {
```

Comment:
Cool, thanks.

Ok, I guess writing it either way would be ok with me. We don't modify the functions, but on the other hand it is a little clutter too.